Tasks/AndroidSigningV3/androidsigning.ts

```diff
@@ -0,0 +1,135 @@
+import * as path from 'path';
+import * as Q from 'q';
+import * as tl from 'vsts-task-lib/task';
+
+const findAndroidTools = (tool: string): string => {
+    const androidHome = tl.getVariable('ANDROID_HOME');
+    if (!androidHome) {
+        throw tl.loc('AndroidHomeNotSet');
+    }
+
+    // add * in search as on Windows the tool may end with ".exe" or ".bat"
+    const toolsList = tl.findMatch(tl.resolve(androidHome, 'build-tools'), tool + '*', null, { matchBase: true });
+
+    if (!toolsList || toolsList.length === 0) {
```

Comment:
dead code -- `toolsList.length === 0` implies `!toolsList`

Reply:
Need some refreshers on dynamic languages...  -_- Wouldn't you hit NRE (or equivalent of it in javascript) if you don't check for `!toolsList`?

Reply:
No you're right.  Just `if (!toolsList)` because an empty list is falsy

Reply:
empty list isn't falsy in my testing:

```
 a = 0
 b = []
 if (!a) { console.log("a is false") }
 if (!b) { console.log("b is false") }
```

prints `a is false`

Reply:
All right, I changed it back to be explicit.  Safety first.  :)

Reply:
Ughhhh @hashtagchris you're right: https://developer.mozilla.org/en-US/docs/Glossary/Truthy

I was confused because:

![image](https://user-images.githubusercontent.com/33549821/41743918-3c8c9cfc-7570-11e8-9d41-4f074d5d5388.png)

And because `[]` is falsy in Python: 

![image](https://user-images.githubusercontent.com/33549821/41743995-7666ec70-7570-11e8-8e70-f4862a4d3837.png)

Reply:
And of course `''` is still falsy, which makes no sense
![image](https://user-images.githubusercontent.com/33549821/41744097-b624f56e-7570-11e8-8951-ded00041047a.png)